proposals/state-machine-restriction-updates.md

```diff
@@ -0,0 +1,61 @@
+# State machine restriction updates
+
+## Summary
+[summary]: #summary
+
+- Allow `using (ref struct)` and `unsafe` blocks in iterators and async methods provided there is no `yield` or `await` inside the blocks.
```

Comment:
It's not so much about `using (ref struct)` as using `ref struct` in general: declaring locals, passing values, etc ... 

Note: as you brought up should consider `ref` locals as well.